### PR TITLE
feat: Add wrappers for ROOT rootcp CLI tool

### PR DESCRIPTION
**Pull request by GoodingJamie**
Adds Snakemake wrappers for the [ROOT command line tool](https://root.cern/manual/root_files/#root-command-line-tools) `rootcp`.

### QC

* [x] I confirm that I have followed the [documentation for contributing to `snakemake-wrappers`](https://snakemake-wrappers.readthedocs.io/en/stable/contributing.html).

While the contributions guidelines are more extensive, please particularly ensure that:
* [x] `test.py` was updated to call any added or updated example rules in a `Snakefile`
* [x] `input:` and `output:` file paths in the rules can be chosen arbitrarily
* [x] wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`)
* [x] temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to 
* [x] the `meta.yaml` contains a link to the documentation of the respective tool or command under `url:`
* [x] conda environments use a minimal amount of channels and packages, in recommended ordering

## Summary by CodeRabbit

- **New Features**
	- Introduced a Conda environment specification for Linux 64-bit to ensure reproducibility.
	- Added a tailored Conda environment configuration for the ROOT framework, enhancing scientific computing capabilities.
	- Launched the `rootcp` tool for copying ROOT files and objects via command line.

- **Bug Fixes**
	- Improved usability and documentation for the `rootcp` tool.

- **Tests**
	- Added a new test case to validate the functionality of the `rootcp` process.

### CODE REVIEW

File: phys/root/rootcp/environment.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+channels:
+  - conda-forge
+dependencies:
+  - ROOT >6.24.0
```
Comment (fgvieira) [inline phys/root/rootcp/environment.yaml:4]:
```suggestion
  - ROOT =6.30.4
```

File: phys/root/rootcp/wrapper.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+__author__ = "Jamie Gooding"
+__copyright__ = "Copyright 2024, Jamie Gooding"
+__email__ = "jamie.gooding@cern.ch"
+__license__ = "MIT"
+
+
+from snakemake.shell import shell
+
+extra = snakemake.params.get("extra", "")
+log = snakemake.log_fmt_shell()
```
Comment (fgvieira) [inline phys/root/rootcp/wrapper.py:10]:
```suggestion
log = snakemake.log_fmt_shell(stdout=True, stderr=True)
```

File: phys/root/rootcp/wrapper.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+object_in = snakemake.params.get("input_object_name", "")
+if object_in:
+    object_in = ":" + object_in
```
Comment (fgvieira) [inline phys/root/rootcp/wrapper.py:14]:
```suggestion
    object_in = f":{object_in}"
```

File: phys/root/rootcp/wrapper.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+object_out = snakemake.params.get("output_object_name", "")
+if object_out:
+    object_out = ":" + object_out
```
Comment (fgvieira) [inline phys/root/rootcp/wrapper.py:18]:
```suggestion
    object_out = f":{object_out}"
```

**Comment by fgvieira** [comment]
Thanks a lot for your contributions!
Just some small cosmetic tweaks. :smile:

File: phys/root/rootcp/wrapper.py (hunk 3 above)
Comment (GoodingJamie) [reply phys/root/rootcp/wrapper.py]: Implemented in https://github.com/snakemake/snakemake-wrappers/pull/3251/commits/0a14defe533b6956c28150457180f78a2a171600

File: phys/root/rootcp/wrapper.py (hunk 4 above)
Comment (GoodingJamie) [reply phys/root/rootcp/wrapper.py]: Implemented in https://github.com/snakemake/snakemake-wrappers/pull/3251/commits/0a14defe533b6956c28150457180f78a2a171600

File: phys/root/rootcp/wrapper.py (hunk 2 above)
Comment (GoodingJamie) [reply phys/root/rootcp/wrapper.py]: Implemented in https://github.com/snakemake/snakemake-wrappers/pull/3251/commits/0a14defe533b6956c28150457180f78a2a171600

**Review by fgvieira — APPROVED** [review]

**Pull request merged**